### PR TITLE
Include discretionary release details

### PR DESCRIPTION
**Pull request by samusgray**
This completes the outstanding retro action item https://trello.com/c/VMDWJRdX

### CODE REVIEW

File: protocol/development-workflow/README.md
Comment (cbortz) [inline protocol/development-workflow/README.md]: An well-placed typo 'transaltation'... I can't tell if this is a bug or a feature!

Comment (KrishnaKulkarni) [reply protocol/development-workflow/README.md]: Hahah. Yeah is this intentional?

File: protocol/development-workflow/README.md
Comment (KrishnaKulkarni) [inline protocol/development-workflow/README.md]: I would add the examples that I think are even more classic: fixing/refactoring tests

File: protocol/development-workflow/README.md
Comment (KrishnaKulkarni) [inline protocol/development-workflow/README.md]: I think this would need to be `git push staging head:master`. While we're here we might even adjust this to reflect our current set up with something like `git push <staging-remote> head:master` since we'll be using `hoth` or `naboo` but not `staging`

**Review by KrishnaKulkarni — APPROVED** [review]
I would not merge without responding to those suggested changes, but I leave it to your discretion to fix those as appropriate and merge. LGTM!

File: protocol/development-workflow/README.md
Comment (samusgray) [reply protocol/development-workflow/README.md]: Hahaha, I'd like to pretend that was intentional.

File: protocol/development-workflow/README.md
Comment (samusgray) [reply protocol/development-workflow/README.md]: Good call.

File: protocol/development-workflow/README.md
Comment (samusgray) [reply protocol/development-workflow/README.md]: I like it.

**Pull request merged**
